api_app/api.py

```diff
@@ -421,6 +423,58 @@ def download_sample(request):
         )
 
 
+@api_view(["PATCH"])
+def kill_running_job(request):
+    """
+    kill running job by closing celery tasks and marking as killed
+    :param request: job_id
+    :returns: 200 if killed, 404 not found, 403 forbidden, 400 bad request
+    """
+    try:
+        data_received = request.query_params
+        logger.info(
+            f"kill_running_job received request from {str(request.user)}."
+            f"Data received {data_received}."
+        )
+        if "job_id" not in data_received:
+            return Response({"error": "821"}, status=status.HTTP_400_BAD_REQUEST)
+        # get job object
+        try:
+            job = models.Job.objects.get(id=data_received["job_id"])
+        except models.Job.DoesNotExist:
+            return Response(
+                {"detail": "job not found"}, status=status.HTTP_404_NOT_FOUND
+            )
+        # check permission
+        if not request.user.has_perm("api_app.change_job", job):
```

Comment:
Regarding this permission, would also need to add an entry for `assign_perm` in the post_migrate signal handler under signal_handler.py

Reply:
I'm waiting for the pr on post migrate signal. I'll change this once it's done. For now, I added the entry in the save one.

Reply:
Imma merge your PR first and make changes accordingly since the tests are failing for the other PR.